toponetx/classes/cell_complex.py

```diff
@@ -2407,6 +2407,40 @@ def get_linegraph(self, s: int = 1, cells: bool = False) -> nx.Graph:
             M = self.node_to_all_cell_adjacnecy_matrix(s=s)
 
         return nx.from_scipy_sparse_array(M)
+    
+    def to_hasse_graph(self) -> nx.DiGraph:
+        """Create Hasse graph of self.
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        nx.DiGraph
+            A NetworkX Digraph representing the Hasse graph of the Cell Complex.
+
+        Examples
+        --------
+        >>> CC = CellComplex()
+        >>> CC.add_cell([1, 2, 3, 4], rank=2)
+        >>> G = CC.to_hasse_graph()
+        """        
+        G = nx.DiGraph()
+        for n in self.nodes:
+            G.add_node(str(tuple((n,))))
+        for e in self.edges:
+            (u,v) = sorted(e)
+            
+            G.add_node( str((u,v)) )
+            G.add_edge(str(tuple((u,))),str((u,v)))
+            G.add_edge(str(tuple((v,))),str((u,v)))
+        for c in self.cells:
+            G.add_node( str(c.elements) )
+            for e in list(c.boundary):
+                (u,v) = sorted(e)
+                G.add_edge(str( (u,v) ),str(c.elements))
+                G.add_edge(str( (u,v) ),str(c.elements))                       
```

Comment:
Is there a reason why you add the tuples as strings? NetworkX is perfectly happy with tuple identifiers:

```python3
G = nx.Graph()
G.add_node((1, 2))
G.add_edge((1, 2), (4, 5))
G.number_of_nodes()  # == 2
G.number_of_edges()  # == 1
```

Reply:
fixed.